frontend/src/app/reports/reports.component.ts

```diff
@@ -0,0 +1,301 @@
+import {Component, OnInit} from '@angular/core';
+import {ReportsService} from './reports.service';
+import * as c3 from 'c3';
+import {ChartAPI} from 'c3';
+import {FormControl, Validators} from "@angular/forms";
+
+@Component({
+  selector: 'app-reports',
+  templateUrl: './reports.component.html',
+  styleUrls: ['./reports.component.css']
+})
+export class ReportsComponent implements OnInit {
+
+    // todo: styles below should be added to styleURLs above when patternfly style is ready
+    // '../../../node_modules/patternfly/dist/css/patternfly.min.css',
+    // '../../../node_modules/patternfly/dist/css/patternfly-additions.css'
+
+        /*
+    /events/count/{timeframe}
+
+    Example API points:
+    localhost:8080/events/count/week
+      {
+      "1st (2017)" : 10,
+      "2nd (2017)": 5,
+      ...
+      "52nd (2017)": 3
+      }
+
+    localhost:8080/events/count/month
+      {
+        "January (2017)": 30,
+        "February (2017)": 40
+      }
+
+    localhost:8080/events/count/year
+      {
+        "2017": 300,
+        "2018": 1000
+      }
+
+    localhost:8080/events/count/quarter
+      {
+      "Q1 (2017)": 100,
+      "Q2 (2017)": 200,
+      "Q3 (2017)": 300,
+      "Q4 (2017)": 400,
+      "Q1 (2018)": 500,
+      "Q2 (2018)": 600
+      }
+
+     */
+
+    // todo: move h2ms-specific stuff to config
+    baseURL = 'http://localhost:8080/';
+    plots = [{value: 'events/count/', viewValue: 'number of observations'}];
+    timeGroupings = [{value: 'year', viewValue: 'year'},
+        {value: 'quarter', viewValue: 'quarter'},
+        {value: 'month', viewValue: 'month'},
+        {value: 'week', viewValue: 'week'}];
+
+    // todo: chart title
+
+    // make false when retrieving data from backend
+    progressBarIsHidden = true;
+
+    // make false when data has been retrieved and json is empty
+    emptyJSONReturned = false;
+
+    // chart related variables
+     chart: ChartAPI;
+
+    /**
+     * form controls allow required fields
+     */
+    plotFormControl = new FormControl('', [
+        Validators.required,
+    ]);
+
+    groupingFormControl = new FormControl('', [
+        Validators.required,
+    ]);
+
+
+    constructor(private reportsService: ReportsService) {}
+
+    ngOnInit() {
+        this.progressBarIsHidden = true;
+        this.emptyJSONReturned = false;
```

Comment:
Kind of strange. You are assigning default values to them here and on lines 65 and 68.

You probably shouldn't assign values on lines 65 and 68 since they are going to be overridden here.